store/tikv/coprocessor.go

```diff
@@ -436,9 +436,10 @@ func (it *copIterator) open(ctx context.Context) {
 		})
 	}
 	taskSender := &copIteratorTaskSender{
-		taskCh: taskCh,
-		wg:     &it.wg,
-		tasks:  it.tasks,
+		taskCh:   taskCh,
+		wg:       &it.wg,
+		tasks:    it.tasks,
+		finished: it.finished,
```

Comment:
The default value of finished is false, is there any chance the value is true?

Reply:
`finished` is a `chan struct{}`, it should be initialized rather than `nil`